client/src/components/Shared/Button/index.js

```diff
@@ -1,5 +1,22 @@
-import React from 'react';
+import React, { Component } from 'react';
+import { withRouter } from 'react-router-dom';
 import StyledButton from './button.style';
 
-const Button = () => <StyledButton type="button">BACK</StyledButton>;
-export default Button;
+
+class Button extends Component {
+  constructor(props) {
```

Comment:
This doesn't have to be a class component. Have a look at the component Hadeel wrote today.